cockatrice/src/carddatabase.cpp

```diff
@@ -340,6 +341,69 @@ void PictureLoader::processLoadQueue()
     }
 }
 
+int PictureLoader::getIdFromHtml(QString html, QString lang)
+{
+    QStringList langs = html.split("class=\"cardItem");
+    for(int i = 1; i < langs.size(); ++i)
+    {
+        const QString langHtml = langs.at(i);
+        if(langHtml.contains(lang,Qt::CaseInsensitive))
+        {
+            int posStart = langHtml.indexOf("multiverseid=") + 13;
+            int posEnd = langHtml.indexOf("\">",posStart);
+            QString sId = langHtml.mid(posStart, posEnd-posStart);
+            return sId.toDouble();
+        }
+    }
+    return 0;
+}
+
+QMap<QString,QString> langDict;
+
+int PictureLoader::getTranslatedId(int currentId)
+{
+    if(langDict.isEmpty())
+    {
+        langDict["cs"] = "繁體中文";
+        langDict["de"] = "Deutsch";
+        langDict["es"] = "Español";
+        langDict["fr"] = "Français";
+        langDict["it"] = "Italiano";
+        langDict["ja"] = "日本語";
+        langDict["ko"] = "한국어";
+        langDict["pt_BR"] = "Português";
+        langDict["pt"] = "Português";
+        langDict["ru"] = "русский язык";
+        langDict["zh-Hans"] = "简体中文";
+    }
+
+    QString lang = langDict.value(settingsCache->getLang(), "English");
+    if(lang == "English")
+        return 0;
+
+    QEventLoop eventLoop;
+
+    QNetworkAccessManager mgr;
+    QObject::connect(&mgr, SIGNAL(finished(QNetworkReply*)), &eventLoop, SLOT(quit()));
+
+    QNetworkRequest req( QUrl( "http://gatherer.wizards.com/Pages/Card/Languages.aspx?multiverseid="+QString::number(currentId,'f',0)) );
```

Comment:
This work should happen in Oracle, I don't think we should do it on the fly.

We could expand cards.xml to support a `lang` attribute on card text, card name, etc, and then cockatrice could select which version to display based on that. We'd set a fallback language of english, and the english cardname would still be the master in terms of what the client sends to the server and what it searches for when getting a name from the server.

Does that make sense?